src/sentry/api/endpoints/organization_prevent_github_repos.py

```diff
@@ -0,0 +1,138 @@
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.constants import ObjectStatus
+from sentry.integrations.services.integration import integration_service
+from sentry.integrations.types import IntegrationProviderSlug
+from sentry.models.organization import Organization
+from sentry.models.repository import Repository
+from sentry.seer.signed_seer_api import sign_with_seer_secret
+
+logger = logging.getLogger(__name__)
+
+
+@region_silo_endpoint
+class OrganizationPreventGitHubReposEndpoint(OrganizationEndpoint):
+    """
+    Lists GitHub orgs linked to the Sentry org enabled for Prevent AI
+
+    GET /organizations/{organization_id_or_slug}/prevent/github/repos/
+    """
+
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.CODECOV
+
+    def _fetch_seer_integrated_repos(self, organization_names: list[str]) -> dict[str, list[str]]:
+        """
+        Fetch repos from Seer for the given GitHub organizations that are integrated with Seer GH app.
+
+        Returns a dict mapping GitHub org name to list of full repo names
+        """
+        path = "/v1/automation/codegen/prevent/integrated-repos"
+        body = orjson.dumps(
+            {
+                "organization_names": organization_names,
+                "provider": "github",
+            }
+        )
+
+        response = requests.post(
+            f"{settings.SEER_AUTOFIX_URL}{path}",
+            data=body,
+            headers={
+                "content-type": "application/json;charset=utf-8",
+                **sign_with_seer_secret(body),
+            },
+        )
+        response.raise_for_status()
+
+        return response.json().get("integrated_repos", {})
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        """
+        List GitHub repositories linked to the Sentry organization for Prevent AI.
+
+        Returns GitHub repos by GitHub orgs for the specified Sentry organization.
+        """
+        logger.info(
+            "PreventGitHubRepos endpoint called",
+            extra={
+                "organization_id": organization.id,
+                "organization_slug": organization.slug,
+                "organization_name": organization.name,
+                "user_id": request.user.id if request.user.is_authenticated else None,
+            },
+        )
+
+        github_org_integrations = integration_service.get_organization_integrations(
+            organization_id=organization.id,
+            providers=[IntegrationProviderSlug.GITHUB.value],
+            status=ObjectStatus.ACTIVE,
+        )
+
+        integration_map = {}
+        for org_integration in github_org_integrations:
+            integration = integration_service.get_integration(
```

Comment:
is there a reason why we can't use `integration_service.get_integrations`? this results in an N+1 RPC issue since we're doing one call per integration.